scatterer/Effects/Proland/Ocean/OceanRenderingHook.cs

```diff
@@ -139,79 +156,21 @@ private void RecreateCommandBuffer()
 
             // then set the textures for the scattering shader
-            rendererCommandBuffer.SetGlobalTexture("ScattererScreenCopy", oceanRenderTexture);
-        }
-
-        private void GetTargetDimensions(out int screenWidth, out int screenHeight)
-        {
-            if (targetCamera.activeTexture)
-            {
-                screenWidth = targetCamera.activeTexture.width;
-                screenHeight = targetCamera.activeTexture.height;
-            }
-            else
-            {
-                screenWidth = Screen.width;
-                screenHeight = Screen.height;
-            }
-        }
-
-        void CreateRenderTextures ()
-        {
-            targetCamera.forceIntoRenderTexture = true; //do this to force the camera target orientation to always match depth orientation
-                                                        //that way we don't have to worry about flipping them separately
-            hdrEnabled = targetCamera.allowHDR;
-
-            GetTargetDimensions(out width, out height);
-            CreateTextures(width, height);
-        }
-
-        private void CreateTextures(int targetWidth, int targetHeight)
-        {
-			if (depthCopyRenderTexture)
-				depthCopyRenderTexture.Release();
-
-			if (oceanRenderTexture)
-				oceanRenderTexture.Release();
-
-			oceanRenderTexture = new RenderTexture(targetWidth, targetHeight, 0, hdrEnabled ? RenderTextureFormat.DefaultHDR : RenderTextureFormat.ARGB32);
-            oceanRenderTexture.useMipMap = false;
-            oceanRenderTexture.autoGenerateMips = false;
-            oceanRenderTexture.Create();
-
-            depthCopyRenderTexture = new RenderTexture(targetWidth, targetHeight, 0, RenderTextureFormat.RFloat);
-            depthCopyRenderTexture.useMipMap = false;
-            depthCopyRenderTexture.autoGenerateMips = false;
-            depthCopyRenderTexture.filterMode = FilterMode.Point;
-            depthCopyRenderTexture.Create();
+            rendererCommandBuffer.SetGlobalTexture("ScattererScreenCopy", oceanRenderTextureNameID);
         }
 
         public void EnableForThisFrame()
 		{
 			if (!renderingEnabled)
             {
-                int targetWidth, targetHeight;
-                GetTargetDimensions(out targetWidth, out targetHeight);
-
-                if (hdrEnabled != targetCamera.allowHDR || width != targetWidth || height != targetHeight || !oceanRenderTexture.IsCreated() || !depthCopyRenderTexture.IsCreated())
+                if (hdrEnabled != targetCamera.allowHDR)
                 {
-                    CreateRenderTextures();
                     RecreateCommandBuffer();
                 }
 
-				bool screenShotModeEnabled = GameSettings.TAKE_SCREENSHOT.GetKeyDown(false);
-				if (oceanScreenShotModeEnabled != screenShotModeEnabled)
-                {
-					// Resize textures
-					int superSizingFactor = screenShotModeEnabled ? Mathf.Max(GameSettings.SCREENSHOT_SUPERSIZE, 1) : 1;

```

Comment:
I am pretty sure that temporary RTs don't detect/work correctly for supersized screenshots. You can leave it like this for now, but I have a branch with new ocean features using tempRTs which I will probably have to check after merging this.

Reply:
I did test this and it seemed fine.

https://imgur.com/e7X5u2l